### PR TITLE
updated scikit learn and added multi-layer perceptron classifier

### PR DESCRIPTION
**Pull request by hitenkoku**
I added multi-layer perceptron classifier.
updated scikit-learn so  scikit-learn version in tram is very old (0.24.1) 

# What Changed

1. update scikit-learn version (0.24.1 -> 1.0.2)
2. add multi-layer perceptron classifier model

### CODE REVIEW

**Comment by emmanvg** [comment]
@m3mike the linter is asking for a bunch of changes to `src/tram/tram/ml/base.py` that are not reported on other PRs. Some of the changes are ok, but others I am not sure of. Should we look into the linter configuration or just apply the requested changes as-is?

**Comment by m3mike** [comment]
> @m3mike the linter is asking for a bunch of changes to `src/tram/tram/ml/base.py` that are not reported on other PRs. Some of the changes are ok, but others I am not sure of. Should we look into the linter configuration or just apply the requested changes as-is?

@emmanvg, the lint checks were added recently and only trigger when a code file has been touched by a PR, which is probably why they haven't been seen much in other PRs yet. 
I don't see any issue with the changes from the linter, so should be fine. 

If you have a Mac/Linux VM/system available, if you run `pre-commit run --file src/tram/tram/ml/base.py` that will automatically run all the appropriate linting/formatting stuff for you. 
If not, let me know and I can quickly just check out this branch, run the linter, and push the change back up to pass CI.

**Comment by emmanvg** [comment]
That's ok, I can run the linter on my end. Just wanted to make sure we are aware about the changes we are about to bring in.

File: src/tram/tram/ml/base.py [hunk 1]
```diff
@@ -1,26 +1,27 @@
-from abc import ABC, abstractmethod
-from datetime import datetime, timezone
-from io import BytesIO
-from os import path
 import pathlib
 import pickle
+import re
 import time
 import traceback
+from abc import ABC, abstractmethod
```
Comment (mehaase) [inline src/tram/tram/ml/base.py:6]: Is there a reason why these imports were re-ordered? They were alpha sorted before, which is preferable.

File: src/tram/tram/ml/base.py [hunk 2]
```diff
@@ -1,26 +1,27 @@
...
 
-from bs4 import BeautifulSoup
-from constance import config
-from django.db import transaction
-from django.conf import settings
 import docx
 import nltk
 import pdfplumber
-import re
+from bs4 import BeautifulSoup
```
Comment (mehaase) [inline src/tram/tram/ml/base.py:14]: Same as above: why re-order these imports?

**Review by mehaase — CHANGES_REQUESTED** [review]

**Comment by mehaase** [comment]
I left some feedback on the PR, but also wanted to comment on the linter. I suggest that we lint the whole repo now and commit it in one go. It's hard to read a PR when 90% of the changes are code formatting to satisfy the linter. Linting the repo will cause merge conflicts with open PRs, of course, but better to rebase those PRs than to clutter the PRs with unrelated changes.

File: src/tram/tram/ml/base.py (hunk 1 above)
Comment (m3mike) [reply src/tram/tram/ml/base.py:6]: it's an isort configuration. each section is alphabetized, and it goes as follows: 

```
import python_builtins

import 3rd party (pypi modules)

import my library
```

and then each section is broken down into `import` statements, and then `from` statements

**Comment by m3mike** [comment]
> I left some feedback on the PR, but also wanted to comment on the linter. I suggest that we lint the whole repo now and commit it in one go. It's hard to read a PR when 90% of the changes are code formatting to satisfy the linter. Linting the repo will cause merge conflicts with open PRs, of course, but better to rebase those PRs than to clutter the PRs with unrelated changes. Cc: @m3mike @emmanvg

RE: linting, that makes sense. I can do that in a sec. 
Also, separate note, in general I don't feel strongly as far as specific lint/format settings, so long as they can be automatically applied by some tool. Completely open to any proposed changes to the linting configuration (mostly in pyproject.toml) if everyone else wants it.

File: src/tram/tram/ml/base.py (hunk 1 above)
Comment (emmanvg) [reply src/tram/tram/ml/base.py:6]: I am in favor of what @m3mike suggests as the preferred approach to sorting imports. 
 - First built-ins
 - 3rd party
 - internal imports

**Comment by emmanvg** [comment]
Agreed, I think its better to take care of the formatting in a single PR and that should help us focus on the real changes to solve the bugs. In general the original changes from this PR were really simple, but the linter artificially made the amount of changes grow.

**Review by mehaase — APPROVED** [review]

**Pull request merged**